### PR TITLE
Set top level directory for legacy make test command

### LINKED ISSUE #4939: unittest discover command in Makefile misconfigured
### Information

- **Qiskit Terra version**: from `master` as of this writing
- **Python version**: 3.8.3
- **Operating system**: macOS 10.15.5

### What is the current behavior?

After running the tests on a clean clone, the following import error occurs, failing the tests:

```
 ======================================================================
ERROR: transpiler.test_faulty_backend (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: transpiler.test_faulty_backend
Traceback (most recent call last):
  File "/Users/andrewmeyer/opt/anaconda3/envs/qiskit-dev/lib/python3.8/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/Users/andrewmeyer/opt/anaconda3/envs/qiskit-dev/lib/python3.8/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/Users/andrewmeyer/code/qiskit-terra/test/python/transpiler/test_faulty_backend.py", line 23, in <module>
    from ..providers.faulty_backends import FakeOurenseFaultyQ1, FakeOurenseFaultyCX01, \
ValueError: attempted relative import beyond top-level package
```

### Steps to reproduce the problem

1. Create/activate a new Anaconda environment:
    ```
    conda create -y -n qiskit-dev python=3
    conda activate qiskit-dev
    ```
1. Follow steps from: https://qiskit.org/documentation/contributing_to_qiskit.html#install-qiskit-terra
1. Run tests
    ```
    make test
    ```

### What is the expected behavior?

All tests should pass

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

Normally for running unittests using tox is recommended [1] but a
Makefile is also available for people who were used to running tests
that way prior to the introduction of the tox config. The python
unittest discovery command set in the legacy test target in the
Makefile was not specifying a top level directory [2], this ends up
causing import error during loading with relative imports that go up a
directory because python can't resolve those if the import goes outside
the start directory. This wasn't an issue in tox or CI because the top
directory is implicitly set by stestr [3] the parallel runner used by
tox and CI. This commit fixes this issue for users of the make target
by setting the -t flag on the unittest discover command to to repo root
so the relative imports can be resolved.

[1] https://github.com/Qiskit/qiskit-terra/blob/master/CONTRIBUTING.md#test
[2] https://docs.python.org/3/library/unittest.html#test-discovery
[3] https://stestr.readthedocs.io/en/latest/MANUAL.html#configuration

### Details and comments

Fixes #4939

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -70,7 +70,7 @@ test_ci:
 	stestr run --concurrency $(CONCURRENCY)
 
 test_randomized:
-	python3 -m unittest discover -s test/randomized -v
+	python3 -m unittest discover -s test/randomized -t . -v
```
Comment (kdk) [inline Makefile:73]: Should this be on L58?

Comment (mtreinish) [reply Makefile:73]: oops yep

Comment (mtreinish) [reply Makefile:73]: Fixed in https://github.com/Qiskit/qiskit-terra/pull/4942/commits/27c7bfa479e477f1238e8681dd3cc5107b74091f

**Review by 1ucian0 — APPROVED** [review]
make test tested testily...

**Pull request merged**